2/contrib/jenkins/install-plugins.sh

```diff
@@ -200,6 +202,17 @@ function doDownload() {
         return 0
     fi
 
+    # Check if the plugin is cached and in correct version; if so, use it instead of downloading
+    # Some plugins do not follow the naming conventions and include the "-plugin" suffix; both cases need to be covered
+    for pluginFilename in "$plugin.jpi" "$plugin-plugin.jpi"; do
+        local cachedPlugin="$INCREMENTAL_BUILD_ARTIFACTS_DIR/plugins/$pluginFilename"
+        if test -f "$cachedPlugin" && unzip -p "$cachedPlugin" META-INF/MANIFEST.MF | tr -d '\r' | grep "^Plugin-Version: ${version}$" > /dev/null; then
```

Comment:
Did you consider using `get_plugin_version()` to get the cached version, and then just comparing that output to `${version}` ?

Or was that related to the further change in install_plugins.sh that you were referring to?  

I'd rather make that work, vs. duplicating/mimicking  the `tr` and `grep` logic

Reply:
I basically just copied the same check which is already there few lines above. `get_plugin_version` is definitely nicer.